docs/requirements.txt

```diff
@@ -1,6 +1,4 @@
-sphinx==3.5.4
-Jinja2==3.0.*
-docutils==0.16
```

Comment:
Jinja2 and docutils are both Sphinx dependencies, we aren't using them directly as far as I know. They were only pinned because Sphinx 3 doesn't work with newer Jinja2 and Sphinx 3–4 don't work with newer docutils. But the latest versions of both should work with Sphinx 5, so there's no longer a reason to pin them.

Reply:
I'm also happy to unpin Sphinx and always test with the latest version, but I know you mentioned wanting more CI stability. None of the other deps are pinned though. Personally, I would only pin things if this file was managed with dependabot and received regular updates.

Reply:
Just pinning sphinx should be ok yes.

Reply:
Let's keep version pinned, otherwise CI/CD might run into a problem every time new version of dependent package is released, which happened  just few days ago with release of protobuf-4.23, see https://github.com/pytorch/pytorch/issues/78362

Reply:
Heh, protobuf was fun.